tests/e2e/helpers.go

```diff
@@ -678,14 +694,31 @@ func suiteErrorMessageMatchesRegex(t *testing.T, f *framework.Framework, namespa
 	return nil
 }
 
-// getNodesWithSelector lists nodes according to a specific selector
-func getNodesWithSelector(f *framework.Framework, labelselector map[string]string) []corev1.Node {
+// getNodesWithSelector lists nodes according to a specific selector.
+func getNodesWithSelector(f *framework.Framework, labelselector map[string]string) ([]corev1.Node, error) {
 	var nodes corev1.NodeList
 	lo := &client.ListOptions{
 		LabelSelector: labels.SelectorFromSet(labelselector),
 	}
-	f.Client.List(goctx.TODO(), &nodes, lo)
-	return nodes.Items
+
+	var lastErr error
+	// retry and ignore errors until timeout
```

Comment:
Was this a linter recommendation? We might want to add to the function comment that it will now wait on results.